### PR TITLE
Sprite and TextureCache from ax::Data 

### PR DESCRIPTION
**Pull request by AlexandreK38**
## Describe your changes
Hi!
This PR is an improvement I made for my game, it allows you to create a Sprite from an Image data (ax::Data) and a key to use for the cache.

### CODE REVIEW

**Comment by paulocoutinhox** [comment]
Hi, very nice. Can you add at least one test?

**Comment by AlexandreK38** [comment]
> Hi, very nice. Can you add at least one test?

Sure, here they are

File: core/2d/Sprite.cpp [hunk 1]
```diff
@@ -304,6 +316,31 @@ bool Sprite::initWithTexture(Texture2D* texture, const Rect& rect, bool rotated)
     return result;
 }
 
+bool Sprite::initWithImageData(const Data& imageData, std::string_view key)
+{
+    if (imageData.getSize() <= 0 || key.empty())
```
Comment (halx99) [inline core/2d/Sprite.cpp]:
```suggestion
    if (imageData.isNull() || key.empty())
```

File: core/renderer/TextureCache.cpp [hunk 2]
```diff
@@ -555,6 +555,58 @@ Texture2D* TextureCache::addImage(Image* image, std::string_view key, PixelForma
     return texture;
 }
 
+Texture2D* TextureCache::addImage(const Data& imageData, std::string_view key)
+{
+    AXASSERT(imageData.getSize() > 0 && key.empty() == false, "TextureCache: imageData MUST not be empty and key not empty");
```
Comment (halx99) [inline core/renderer/TextureCache.cpp]:
```suggestion
    AXASSERT(!imageData.isNull() && !key.empty(), "TextureCache: imageData MUST not be empty and key not empty");
```

**Review by halx99 — CHANGES_REQUESTED** [review]

**Review by halx99 — APPROVED** [review]

**Pull request merged**